File: 452. Minimum Number of Arrows to Burst Balloons/findMinArrowShots.js
/**
 * @param {number[][]} points
 * @return {number}
 */
const findMinArrowShots = (points) => {
  points.sort((a, b) => a[1] - b[1]);
  let amountOfArrows = 0;
  let arrowX = -Infinity;

  for (let i = 0; i < points.length; i++) {
    const [start, end] = points[i];
    if (start > arrowX) {
      amountOfArrows++;
      arrowX = end;
    }
  }

  return amountOfArrows;
};

const points = [[10, 16], [2, 8], [1, 6], [7, 12]];
console.log('findMinArrowShots: ', findMinArrowShots(points));

File: 452. Minimum Number of Arrows to Burst Balloons/findMinArrowShots.py
from typing import List

class Solution:
  def findMinArrowShots(self, points: List[List[int]]) -> int:
    points.sort(key = lambda x: x[1])
    amountOfArrows = 0
    arrowX = float('-inf')

    for point in points:
      [start, end] = point
      if start > arrowX:
        amountOfArrows += 1
        arrowX = end

    return amountOfArrows


points = [[10, 16], [2, 8], [1, 6], [7, 12]]
solution = Solution()
print('findMinArrowShots: ', solution.findMinArrowShots(points))
